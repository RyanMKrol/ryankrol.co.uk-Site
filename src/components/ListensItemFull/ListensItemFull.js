import React, { Component } from 'react';

import styles from './ListensItemFull.module.css';

/**
 * A component representing all information about a "listens" item
 */
class ListensItemFull extends Component {
  /**
   * @class
   * @param {object} props arguments to class
   */
  constructor(props) {
    super();

    this.albumLink = props.albumLink;
    this.albumName = props.albumName;
    this.thumbnail = props.thumbnail;
    this.artist = props.artist;
    this.playcount = props.playcount;

    if (!this.albumLink || !this.albumName || !this.thumbnail || !this.artist || !this.playcount) {
      throw new Error('Could not create ListensItemFull');
    }
  }

  /**
   * Render method
   */
  render() {
    return (
      <div className={styles.listensItem}>
        <a href={this.albumLink}>
          <div className={styles.listensItemImage}>
            <img alt={this.albumName} src={this.thumbnail} />
          </div>
          <span className="highlighted">Artist:</span>
          <span>{this.artist}</span>
          <br />
          <span className="highlighted">AlbumName:</span>
          <span>{this.albumName}</span>
          <br />
          <span className="highlighted">Playcount:</span>
          <span>{this.playcount}</span>
          <br />
        </a>
      </div>
    );
  }
}

export default ListensItemFull;
