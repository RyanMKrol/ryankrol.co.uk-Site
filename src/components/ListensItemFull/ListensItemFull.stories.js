import React from 'react';

import DEFAULT_MOCK from './__mocks__';

export default {
  title: 'Components/ListensItemFull',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the ListensItemFull
 */
export const normal = () => DEFAULT_MOCK;
