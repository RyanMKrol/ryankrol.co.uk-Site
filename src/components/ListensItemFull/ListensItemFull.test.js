import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ListensItemFull from './ListensItemFull';

import DEFAULT_MOCK from './__mocks__';

it('renders ListensItemFull without issue', () => {
  const listensItem = shallow(DEFAULT_MOCK);
  expect(toJson(listensItem)).toMatchSnapshot();
});

it('renders ListensItemFull without throwing', () => {
  expect(() => mount(DEFAULT_MOCK)).not.toThrow();
});

it('ListensItemFull throws without album link', () => {
  expect(() => mount(<ListensItemFull />)).toThrow();
});

it('ListensItemFull throws without album name', () => {
  expect(() => mount(<ListensItemFull albumLink="test-link" />)).toThrow();
});

it('ListensItemFull throws without thumbnail', () => {
  expect(() => mount(<ListensItemFull albumLink="test-link" albumName="test-name" />)).toThrow();
});

it('ListensItemFull throws without artist', () => {
  expect(() => mount(
      <ListensItemFull thumbnail="test-thumbnail" albumLink="test-link" albumName="test-name" />,
  )).toThrow();
});

it('ListensItemFull throws without playcount', () => {
  expect(() => mount(
      <ListensItemFull
        thumbnail="test-thumbnail"
        playcount={123}
        albumLink="test-link"
        albumName="test-name"
      />,
  )).toThrow();
});
