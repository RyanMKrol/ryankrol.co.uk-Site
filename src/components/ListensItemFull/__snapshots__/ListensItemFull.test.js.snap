// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListensItemFull without issue 1`] = `
<div
  className="listensItem"
>
  <a
    href="https://www.last.fm/music/Kendrick+Lamar/DAMN."
  >
    <div
      className="listensItemImage"
    >
      <img
        alt="DAMN"
        src="https://lastfm.freetls.fastly.net/i/u/300x300/243a7444e2d59ff9b38e123f7267e3be.jpg"
      />
    </div>
    <span
      className="highlighted"
    >
      Artist:
    </span>
    <span>
      Kendrick Lamar
    </span>
    <br />
    <span
      className="highlighted"
    >
      AlbumName:
    </span>
    <span>
      DAMN
    </span>
    <br />
    <span
      className="highlighted"
    >
      Playcount:
    </span>
    <span>
      123
    </span>
    <br />
  </a>
</div>
`;
